pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTaskRestletResource.java

```diff
@@ -346,6 +347,29 @@ public StringResultResponse getTaskStateDeprecated(
     }
   }
 
+  @GET
+  @Path("/tasks/all")
+  @Authenticate(AccessType.READ)
+  @ApiOperation("Read all tasks status")
+  public Map<String, TaskState> getAllTasks() {
+    Map<String, TaskState> taskStates = new HashMap<>();
+    _pinotHelixTaskResourceManager.getTaskTypes()
+        .forEach(taskType -> taskStates.putAll(_pinotHelixTaskResourceManager.getTaskStates(taskType)));
+    return taskStates;
+  }
+
+  @POST
+  @Path("/tasks/create")
```

Comment:
I think what you want to do is to {{run}} a specific task type that has already been defined. Correct? If so, can we change the API to call it "run" (or "execute")?

Reply:
sure